SG14/ring.h

```diff
@@ -108,7 +108,15 @@ namespace sg14
 		template <bool C>
 		bool operator==(const ring_iterator<Ring, C>& rhs) const noexcept;
 		template <bool C>
+		bool operator!=(const ring_iterator<Ring, C>& rhs) const noexcept;
+		template <bool C>
 		bool operator<(const ring_iterator<Ring, C>& rhs) const noexcept;
```

Comment:
Hm. So in a ringbuffer, you'd say it was *not* necessarily true that `rv.end() - rv.begin() == rv.size()`?